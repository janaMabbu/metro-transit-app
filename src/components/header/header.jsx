import React from 'react'
import './header.less'

export const Header = () => {
  return (
      <div className="col-sm-12 app-header__content">
        <div className="app-header__logo"></div>
        <div className="app-header__banner" >
          <h1 className="app-header__text">Ready for NexTrip?</h1>
        </div>
      </div>
    )
}