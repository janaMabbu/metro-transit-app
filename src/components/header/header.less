@import "~less/common.less";

.app-header {
  &__content {
    background-color: @background-color;
  }

  &__logo {
    background: url('../../images/MetroTransitLogo.svg') no-repeat center;
    height: 30px;
    width: 200px;
    background-size: contain;
    margin: 20px;
  }

  &__banner {
    background: url('../../images/nextrip.jpg') no-repeat center;
    background-size: cover;
    background-position: 90%;
    background-repeat: no-repeat;
    height: 13rem;
    position: relative;
  }

  &__text {
    position: absolute;
    background-color: #fff;
    color: #8a8b8a;
    opacity: 0.9;
    margin-bottom: 0;
    padding: 1rem;
    width: auto;
    font-size: 2rem !important;
    left: 20px;
    bottom: 20px;
  }

}