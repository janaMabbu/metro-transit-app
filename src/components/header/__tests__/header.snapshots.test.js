import { shallow } from 'enzyme'
import React from 'react'

import { Header } from 'components/header/header'

describe('Component - Header', () => {
  it('should render the Header snapshot', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
  })

  it('it should contain logo , banner, and text', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper.find('.app-header__logo').length).toBe(1)
    expect(wrapper.find('.app-header__banner').length).toBe(1)
    expect(wrapper.find('.app-header__text').length).toBe(1)
  })

})