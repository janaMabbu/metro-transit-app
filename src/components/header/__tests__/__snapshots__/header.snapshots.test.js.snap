// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Header should render the Header snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="app-header__logo"
        />,
        <div
          className="app-header__banner"
        >
          <h1
            className="app-header__text"
          >
            Ready for NexTrip?
          </h1>
        </div>,
      ],
      "className": "col-sm-12 app-header__content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "app-header__logo",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="app-header__text"
          >
            Ready for NexTrip?
          </h1>,
          "className": "app-header__banner",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ready for NexTrip?",
            "className": "app-header__text",
          },
          "ref": null,
          "rendered": "Ready for NexTrip?",
          "type": "h1",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="app-header__logo"
          />,
          <div
            className="app-header__banner"
          >
            <h1
              className="app-header__text"
            >
              Ready for NexTrip?
            </h1>
          </div>,
        ],
        "className": "col-sm-12 app-header__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "app-header__logo",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="app-header__text"
            >
              Ready for NexTrip?
            </h1>,
            "className": "app-header__banner",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ready for NexTrip?",
              "className": "app-header__text",
            },
            "ref": null,
            "rendered": "Ready for NexTrip?",
            "type": "h1",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
